dateparser/utils.py

```diff
@@ -99,25 +130,15 @@ def constructor(cls, *args, **kwargs):
             registry_dict = getattr(cls, "__registry_dict")
 
             if key not in registry_dict:
-                registry_dict[key] = creator(cls, *args, **kwargs)
+                registry_dict[key] = creator(cls, *args)
                 setattr(registry_dict[key], 'registry_key', key)
             return registry_dict[key]
         return staticmethod(constructor)
 
-    def self_destruct(initializer):
-        def destructor(self, *args, **kwargs):
-            if getattr(self, "__initialized", False):
-                return
-            else:
-                initializer(self, *args, **kwargs)
-                setattr(self, "__initialized", True)
-        return destructor
-
     if not (hasattr(cls, "get_key")
```

Comment:
this wasn't needed once `Language` classes stopped to be returning from registry. Hence, removed.